python/enthic/compute_stats.py

```diff
@@ -0,0 +1,234 @@
+            tree_item['data']['status'] = "checked"
+            # Fix children values if needed
+            for child_name in tree_item['children']:
+                child_data = tree_item['children'][child_name]['data']
+                if math.isnan(child_data['value']):
+                    child_data['status'] = "computed"
+                    if 'computedValue' in child_data:
+                        child_data['value'] = child_data['computedValue']
+                    else:
+                        child_data['value'] = 0
+
+            computed_sum = computed_sum_from_computed
+        # If there is only on value missing from children, set this child's value equal to the computed difference
+        elif child_missing_count == 1:
+            for child_name in tree_item['children']:
+                child_data = tree_item['children'][child_name]['data']
+                if math.isnan(child_data["value"]):
+                    child_data['computedValue'] = (value - computed_sum) / child['sign']
+                    child_data['value'] = child_data['computedValue']
+                    child_data['status'] = "computed"
+                    tree_item['data']['status'] = "checked"
+                    break
+
+        # If official value match computed value by adding all children's value (no substraction) with less than 0.5% error
+        elif abs((computed_sum_without_sign - value) / value) < relative_error or abs(computed_sum_without_sign - value) < absolute_error:
+            tree_item['data']['status'] = "checked"
+            # Fix children sign and/or set to zero missing values if any
+            for child_name in tree_item['children']:
+                flip_sign(tree_item['children'][child_name])
+                set_to_zero_computed(tree_item['children'][child_name])
+        elif abs((computed_sum - value) / value) > 100:
+            tree_item['data']['status'] = "error"
+        else:
+            tree_item['data']['status'] = "error"
+
+    if computed_sum != tree_item['data']['value']:
+        tree_item['data']['computedValue'] = computed_sum_from_computed
+
+def set_to_zero_computed(tree_item):
+    """
+    set value of given item and it's children to zero when it hasn't another value
+        :param tree_item : item to set to zero
+    """
+    if math.isnan(tree_item['data']['value']) and ('computedValue' not in tree_item['data'] or tree_item['data']['computedValue'] == 0):
+        tree_item['data']['value'] = 0
+        tree_item['data']['status'] = "computed"
+        if 'children' in tree_item:
+            for child_name in tree_item['children']:
+                set_to_zero_computed(tree_item['children'][child_name])
+
+def flip_sign(item):
+    """
+    Flip sign of given item
+        :param item : item to flip
+    """
+    if item['sign'] == -1:
+        item['data']['value'] = -item['data']['value']
+        item['data']['status'] = "signFlipped"
+
```

Comment:
PEP 8: E302 expected 2 blank lines, found 1, use autopep 8 or an IDE that reformat according to PEP8 e.g. Pycharm Ctrl+Alt+L .